lib/sensu-plugin/cli.rb

```diff
@@ -44,14 +44,12 @@ def run
       @@autorun = self
       class << self
         def method_added(name)
-          if name == :run
-            @@autorun = self
-          end
+          @@autorun = self if name == :run
         end
       end
 
       at_exit do
-        exit 3 if $!
+        exit 3 if $ERROR_INFO
```

Comment:
using this which I think is good needs `require 'English'` at the top of the file.